### PR TITLE
Rowan/feature/add clippy and warns as errs

### PR DESCRIPTION
**Pull request by rrasmuss4200**
Applied some clippy suggestions and made pushes run clippy and cargo test when building

### CODE REVIEW

File: ex3_ground_station/DFGM/data
Comment (kulovac) [inline ex3_ground_station/DFGM/data:1]: I don't think you meant to push the DFGM and IRIS data files.

**Comment by kulovac** [comment]
There are a lot of Clippy suggestions still, do you want to leave those or fix them now? And none of the tests failed.

File: ex3_ground_station/DFGM/data
Comment (rrasmuss4200) [reply ex3_ground_station/DFGM/data:1]: Thanks for pointing this out. It was in my .gitignore which is weird

**Comment by rrasmuss4200** [comment]
No, the clippy warnings are ok for now. It's just the build warnings we want to keep mind of

File: ex3_shared_libs/interfaces/ipc/src/lib.rs
Comment (kulovac) [inline ex3_shared_libs/interfaces/ipc/src/lib.rs:1]: I get that you could easily remove the commented code, but why leave it there? You could always go into the history to get the code back, right? I'm just wondering if that's a deliberate choice

Comment (rrasmuss4200) [reply ex3_shared_libs/interfaces/ipc/src/lib.rs:1]: You're right. It'd be cleaner just to delete it.

**Review by kulovac — APPROVED** [review]
Seems like everything still works and I don't see any build warnings. I'm definitely a fan of the Clippy/lint addition. Looks good.

**Pull request merged**